module/users/helper.js

```diff
@@ -5,642 +5,762 @@
- const solutionsHelper = require(MODULES_BASE_PATH + "/solutions/helper");
- const observationsHelper = require(MODULES_BASE_PATH + "/observations/helper");
- const submissionsHelper = require(MODULES_BASE_PATH + "/submissions/helper");
- const entitiesHelper = require(MODULES_BASE_PATH + "/entities/helper");
- const observationSubmissionsHelper = require(MODULES_BASE_PATH + "/observationSubmissions/helper");
- const surveysHelper = require(MODULES_BASE_PATH + "/surveys/helper");
- const userExtensionsHelper = require(MODULES_BASE_PATH + "/userExtension/helper");
- const surveySubmissionsHelper = require(MODULES_BASE_PATH + "/surveySubmissions/helper");
- const shikshalokamHelper = require(MODULES_BASE_PATH + "/shikshalokam/helper");
-
+// Dependencies
+
+const entityAssessorsHelper = require(MODULES_BASE_PATH +
+  "/entityAssessors/helper");
+const programsHelper = require(MODULES_BASE_PATH + "/programs/helper");
+const solutionsHelper = require(MODULES_BASE_PATH + "/solutions/helper");
+const observationsHelper = require(MODULES_BASE_PATH + "/observations/helper");
+const submissionsHelper = require(MODULES_BASE_PATH + "/submissions/helper");
+const entitiesHelper = require(MODULES_BASE_PATH + "/entities/helper");
+const observationSubmissionsHelper = require(MODULES_BASE_PATH +
+  "/observationSubmissions/helper");
+const surveysHelper = require(MODULES_BASE_PATH + "/surveys/helper");
+const userExtensionsHelper = require(MODULES_BASE_PATH +
+  "/userExtension/helper");
+const surveySubmissionsHelper = require(MODULES_BASE_PATH +
+  "/surveySubmissions/helper");
+const shikshalokamHelper = require(MODULES_BASE_PATH + "/shikshalokam/helper");
 
 /**
-    * UserHelper
-    * @class
-*/
+ * UserHelper
+ * @class
+ */
 module.exports = class UserHelper {
+  /**
+   * Delete user information.
+   * @method
+   * @name userDelete
+   * @param {Objcet} kafkaEvent
+   * @returns {Object} Object with status if found and updated then will return true else will return false.
+   */
+  static userDelete(kafkaEvent) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let userId = kafkaEvent.edata.userId;
+        let userSurveySubmissionsProfileUpdateData = [];
+        let userObservationProfileUpdateData = [];
+        let userObservationSubmissionProfileUpdateData = [];
+        let updateObject = {
+          $set: { "userProfile.firstName": "Deleted User" },
+          $unset: {
+            "userProfile.email": 1,
+            "userProfile.maskedEmail": 1,
+            "userProfile.maskedPhone": 1,
+            "userProfile.recoveryEmail": 1,
+            "userProfile.phone": 1,
+            "userProfile.lastName": 1,
+            "userProfile.prevUsedPhone": 1,
+            "userProfile.prevUsedEmail": 1,
+            "userProfile.recoveryPhone": 1,
+          },
+        };
+        let userSurveySubmissionData =
+          await surveySubmissionsHelper.surveySubmissionDocuments(
```

Comment:
@ankitpws Use promise.all and call functions within the modules directly that will delete the users and return success along with rows updated.

Reply:
used promise all